src/index.js

```diff
@@ -0,0 +1,143 @@
+ *        4. Immutable. If the value changes between paged queries, it could appear twice.
+ *      The default is to use the Mongo built-in '_id' field, which satisfies the above criteria.
+ *      The only reason to NOT use the Mongo _id field is if you chose to implement your own ids.
+ *    -next {String} The value to start querying the page. Default to start at the beginning of
+ *      the collection.
+ *    -previous {String} The value to start querying previous page. Default is to not query backwards.
+ * @param {Function} done Node errback style function.
+ */
+function find(collection, params, done) {
+  params = _.defaults(params, {
+    query: {},
+    limit: 100,
+    paginatedField: '_id'
+  });
+
+  if (params.previous) params.previous = urlSafeDecode(params.previous);
+  if (params.next) params.next = urlSafeDecode(params.next);
+
+  // The query must always include the paginatedField so know if to get the
+  var fields;
+
+  if (params.fields) {
+    fields = _.extend({}, params.fields, {
+      [params.paginatedField]: 1
+    });
+  }
+
+  if (params.next) {
+    params.query[params.paginatedField] = {
+      $lt: params.next
+    };
+  } else if (params.previous) {
+    params.query[params.paginatedField] = {
+      $gt: params.previous
+    };
+  }
+
+  collection
+    .find(params.query, fields)
+    .sort({
+      [params.paginatedField]: params.previous ? 1 : -1
+    })
+    .limit(params.limit)
+    .toArray((err, results) => {
+      if (err) {
+        done(err);
+        return;
+      }
+
+      // If we sorted reverse to get the previous page, correct the sort order.
+      if (params.previous) results = results.reverse();
+
+      // If the user didn't include the paginated field in their desired fields and we included
+      // it for them, remove it.
+      if (params.fields && !_.has(params.fields, params.paginatedField)) {
+        results = _.map(results, result => _.omit(result, params.paginatedField));
+      }
+
+      // Return 'previous' and 'next' cursors, based on what we know about the results.
+      // This logic is imperfect because it might return 'next' if there are no more results.
+      // This is probably fine since it just means they'll get back an empty query.
+
+      var fullPageOfResults = (results.length === params.limit);
+
+      var response;
+      if (results.length === 0) {
+        if (params.next) {
+          response = {
+            results: [],
+            previous: params.next
+          };
+        } else if (params.previous) {
+          response = {
+            results: [],
+            next: params.previous
+          };
+        } else {
+          response = {
+            results: []
+          };
+        }
+      } else if (fullPageOfResults && !params.next && !params.previous) {
+        response = {
+          results,
+          next: results[results.length - 1][params.paginatedField]
```

Comment:
The paginated field may be gone by this time: https://github.com/mixmaxhq/mongo-cursor-pagination/pull/1/files#diff-1fdf421c05c1140f6d71444ea2b27638R73